perfkitbenchmarker/packages/ycsb.py

```diff
@@ -130,6 +144,26 @@ def _Install(vm):
   vm.RemoteCommand(('mkdir -p {0} && curl -L {1} | '
                     'tar -C {0} --strip-components=1 -xzf -').format(
                         YCSB_BUILD_DIR, YCSB_TAR_URL))
+
+  settings_file = ".m2/settings.xml"
+  proxy_enable = False
```

Comment:
You've got quite a lot of separate state in variables here. Alternatively, how about something like this:

```python
proxy_nodes = []
if FLAGS.http_proxy:
  proxy_nodes.append(CreateProxyElement('http', FLAGS.http_proxy))
...
if proxy_nodes:
  root = Element('settings')
  proxies = SubElement(root, 'proxies')
  proxies.extend(proxy_nodes)
  ...
```

For better isolation, would be nice to move this into a CreateSettingsFile method.